frontend/lib/map/full_map.dart

```diff
@@ -1,26 +1,43 @@
+import 'dart:convert';
+import 'dart:math';
+
 import 'package:flutter/material.dart';
 import 'package:mapbox_gl/mapbox_gl.dart';
 
+typedef void OnFeatureClickCallback(dynamic feature);
+
 class FullMapState extends State<FullMap> {
-  final String mapboxToken;
+  final OnFeatureClickCallback onFeatureClick;
+  final String _mapboxToken;
+  MapboxMapController _controller;
 
-  FullMapState(this.mapboxToken);
+  FullMapState(this._mapboxToken, this.onFeatureClick);
 
   @override
   Widget build(BuildContext context) {
     return new MapboxMap(
-      accessToken: this.mapboxToken,
+      accessToken: this._mapboxToken,
       initialCameraPosition: const CameraPosition(target: LatLng(0.0, 0.0)),
       styleString: "mapbox://styles/elkei24/ckhyn5h2l1yq519r9g3fbsogj",
+      onMapCreated: (controller) => {this._controller = controller},
+      onMapClick: this.onMapClick,
     );
   }
+
+  void onMapClick(Point<double> point, LatLng coordinates) {
+    this._controller.queryRenderedFeatures(point, [], null).then((features) => {
+          if (features?.isNotEmpty)
+            {this.onFeatureClick(json.decode(features[0]))}
```

Comment:
is features[0] here a plain geojson object?

Reply:
I think so. Sadly, the type is specified as `dynamic` (like any, Object), but at runtime it is a string containing exactly the GeoJSON entry